### PR TITLE
fix(babel-parser): Fix end of `range` of `SequenceExpression`

### PR DESCRIPTION
**Pull request by sosukesuzuki**
| Q                        | A 
| ------------------------ | ---
| Patch: Bug Fix?          | Y
| Tests Added + Pass?      |  Y
| License                  | MIT

Caused by #13521.

`range[1]` of `SequenceExpression` is 1 more than expected.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/13731"></a>

### CODE REVIEW

**Review by maxtrem271991 — APPROVED** [review]

**Review by fedeci — APPROVED** [review]

File: packages/babel-parser/src/parser/expression.js [hunk 1]
```diff
@@ -1708,6 +1708,7 @@ export default class ExpressionParser extends LValParser {
       this.finishNode(val, "SequenceExpression");
       val.end = innerEndPos;
       val.loc.end = innerEndLoc;
+      if (this.options.ranges) val.range[1] = innerEndPos;
```
Comment (JLHwung) [inline packages/babel-parser/src/parser/expression.js:1711]: We can call `resetEndLocation` on `val`.

**Review by JLHwung — APPROVED** [review]

**Review by jridgewell — APPROVED** [review]

**Review by maxtrem271991 — APPROVED** [review]
add

File: packages/babel-parser/src/parser/expression.js (hunk 1 above)
Comment (sosukesuzuki) [reply packages/babel-parser/src/parser/expression.js]: Thanks 6f2ae6c

**Review by nicolo-ribaudo — APPROVED** [review]

**Comment by nicolo-ribaudo** [comment]
The CI failure is not related

**Pull request merged**